### ISSUE TITLE
Pkg Request: Suckless Surf - A Keyboard-Driven Alternative to Netsurf

### ISSUE THREAD

**Issue by VehementHam** [oasislinux/oasis#85]
I know Suckless Surf is a litte bloated (moreso than netsurf) because of the web engine, but it is suckless, so take that for what it is worth.

#45 requested to move to Visurf because it supports the Vim bindings. It was rejectied because visurf is bloated.

The most minimal option that I could find for a vim-like browser is Surf. There is a [modal](https://surf.suckless.org/patches/modal/) patch that gives it an insert mode. 

For hints there is a [JS script](https://surf.suckless.org/files/easy_links/)

With these features, Surf is a minimal vim-like alternative to Netsurf.

My request is that it is packaged in addition to netsurf, rather than replacing it.

**Comment by apprehensions** [comment]
Use netsurf. A browser like surf required WebKit which I imagine will be a pain in the ass to package for oasis, at least try packaging it yourself first or understand how massive WebKit is.

**Comment by VehementHam** [comment]
Netsurf is unusable with only the keyboard. Are there any other vim-like browsers that are minimal, and do not use WebKit?

**Comment by VehementHam** [comment]
TBH packaging Surf is not a big deal to me. WebKit would be really nice. I might look into packaging it myself as you suggested.

**Comment by michaelforney** [comment]
I think you are severely underestimating the difficulty of packaging WebKit. For one, it is C++, which goes against the goal of being able to compile the system with cproc or another simple C compiler. It also has *tons* of other dependencies (probably around 20-50 would be needed) which would need to be packaged as well. WebKit, and any other browser framework that works well with modern websites are *huge* (and I can't overstate this) projects.

My suggestion is to build WebKit through pkgsrc, or install it with some other packaging system. Personally, I've started using chromium installed through flatpak, and haven't looked back. It works well with swc, and is much faster and more responsive than any webkit browser I've built myself, and it only takes seconds to install. I have decided that trying to build any modern browser on my own computer is a waste of time. It takes forever even on a modern system, requires a ton of disk space, and you'll often run into a myriad of build failures which require considerable effort to resolve.
